packages/flutter_tools/lib/src/project.dart

```diff
@@ -467,7 +467,11 @@ class AndroidProject extends FlutterProjectPlatform {
   static final RegExp _androidNamespacePattern = RegExp('android {[\\S\\s]+namespace\\s*=?\\s*[\'"](.+)[\'"]');
   static final RegExp _applicationIdPattern = RegExp('^\\s*applicationId\\s*=?\\s*[\'"](.*)[\'"]\\s*\$');
   static final RegExp _imperativeKotlinPluginPattern = RegExp('^\\s*apply plugin\\:\\s+[\'"]kotlin-android[\'"]\\s*\$');
-  static final RegExp _declarativeKotlinPluginPattern = RegExp('^\\s*id\\s+[\'"]kotlin-android[\'"]\\s*\$');
+  static final List<RegExp> _declarativeKotlinPluginPatterns = <RegExp>[
+      RegExp('^\\s*id\\s*\\(?\\s*[\'"]kotlin-android[\'"]\\s*\\)?\\s*\$'),
+      RegExp('^\\s*id\\s*\\(?\\s*[\'"]org.jetbrains.kotlin.android[\'"]\\s*\\)?\\s*\$'),
+  ];
+
```

Comment:
[Here's a small DartPad playground](https://dartpad.dev/?id=3e3acc7072aeff2e4479f2884a79075a) I made to ensure I got the regexes right.